File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task1-Server/src/main/java/com/pet/hotel/data/database/entities/UserEntity.java
package com.pet.hotel.data.database.entities;

import com.pet.hotel.data.enums.UserType;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "user")
@NoArgsConstructor
@Component
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "user_id")
    @Setter(AccessLevel.PUBLIC)
    private int userId;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "password", nullable = false)
    private String passwordHash;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "birth_date")
    private LocalDate birthDate;

    @Column(name = "phone_number", nullable = false, unique = true)
    private String phoneNumber;

    @Column(name = "photo_link")
    private String photoLink;

    @Enumerated(EnumType.STRING)
    @Column(name = "type_user", nullable = false)
    private UserType typeUser;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "user")
    private List<PetEntity> pets;

    public UserEntity(String fullName, String passwordHash, String email, LocalDate birthDate, String phoneNumber, String photoLink, UserType typeUser, List<PetEntity> listOfPets) {
        this.fullName = fullName;
        this.passwordHash = passwordHash;
        this.email = email;
        this.birthDate = birthDate;
        this.phoneNumber = phoneNumber;
        this.photoLink = photoLink;
        this.typeUser = typeUser;
        this.pets = listOfPets;
    }
}

File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task4-MobileClient/src/main/java/com/animal/hotel/data/user/entities/UserEntity.kt
package com.animal.hotel.data.user.entities

import android.os.Parcelable
import com.animal.hotel.utils.enums.UserType
import com.animal.hotel.domain.models.user.User
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.time.LocalDate


data class UserEntity(
    val userId: Int,
    val fullName: String,
    val password: String,
    val email: String,
    val birthDate: LocalDate,
    val phoneNumber: String,
    val typeUser: UserType,
    val photoLink: String?
) {

    fun toUser(): User {
        return User(userId, fullName, password, email, birthDate, phoneNumber, typeUser, photoLink)
    }

    companion object {
        fun toUserEntity(user: User) = UserEntity (
            user.userId,
            user.fullName,
            user.password,
            user.email,
            user.birthDate,
            user.phoneNumber,
            user.typeUser,
            user.photoLink
        )
    }
}
